### ISSUE TITLE
Missing “Boost ECC” flag causing discrepancy with other libraries

### ISSUE THREAD

**Issue by sunknudsen** [zpao/qrcode.react#356]
qrcode.react  uses https://www.nayuki.io/page/qr-code-generator-library under the hood.

I was investigating visual discrepancy between output of qrcode.react vs `CIFilter.qrCodeGenerator()` in Swift for same error correction level.

The discrepancy is caused by “Boost ECC” flag being enabled by default.

I believe “Boost ECC” flag should be user-configurable in qrcode.react.

**Comment by sunknudsen** [comment]
I believe following would make `qrcode.react`  follow reference implementations rather than enable off-spec `boostEcl` by default.

See [https://github.com/nayuki/QR-Code-generator/blob/master/typescript-javascript/qrcodegen.ts?ts=4#L81 and ](https://github.com/nayuki/QR-Code-generator/blob/42a886d7845283a048423bd78a82912584325476/typescript-javascript/qrcodegen.ts#L81) and https://github.com/nayuki/QR-Code-generator/blob/42a886d7845283a048423bd78a82912584325476/typescript-javascript/qrcodegen.ts#L90.

Replacing:

```typescript
return qrcodegen.QrCode.encodeSegments(
  segments,
  ERROR_LEVEL_MAP[level],
  minVersion
);
```

With:

```typescript
return qrcodegen.QrCode.encodeSegments(
  segments,
  ERROR_LEVEL_MAP[level],
  minVersion,
  undefined,
  undefined,
  false
);
```

**Comment by sunknudsen** [comment]
Looks like different libraries also don’t use same mask pattern.

**Comment by zpao** [comment]
Interesting. I did notice that the output was different when I switched to using this library. At the end of the day, everything I tested with scanned so I didn't see a problem in practice.

What's the practical impact here? I'm not opposed to making changes, and honestly may opt to disable the boost than expose another prop.

FWIW I know there's already incomplete behavior (eg not supporting optimized kanji encoding) and I document that in the readme. I'm working on improved docs for v4 too, so would document anything else like this.
